package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,331 @@
+
+
+References
+----------
+
+.. [Maginn2019] Maginn, E. J., Messerly, R. A., Carlson, D. J.; Roe, D. R., Elliott, J. R. Best Practices for Computing Transport Properties 1. Self-Diffusivity and Viscosity from Equilibrium Molecular Dynamics [Article v1.0]. Living J. Comput. Mol. Sci. 2019, 1 (1).
+
+
+
+Classes and Functions
+---------------------
+
+.. autoclass:: EinsteinMSD
+    :members:
+
+"""
+
+from __future__ import division, absolute_import
+
+import numpy as np
+import logging
+import tidynamics
+from ..due import due, Doi
+from .base import AnalysisBase
+
+due.cite(Doi("10.21105/joss.00877"),
+         description="Mean Squared Displacements with tidynamics",
+         path="MDAnalysis.analysis.msd",
+         cite_module=True)
+due.cite(Doi("10.1051/sfn/201112010"),
+         description="FCA fast correlation algorithm",
+         path="MDAnalysis.analysis.msd",
+         cite_module=True)
+del Doi
+
+class EinsteinMSD(AnalysisBase):
+    r"""Class to calculate Mean Squared Displacement by the Einstein relation.
+    
+    Attributes
+    ----------
+    dim_fac : int
+        Dimensionality :math:`d` of the MSD.
+    timeseries : :class:`np.ndarray`
+        The averaged MSD with respect to lag-time.
+    msd_per_particle : :class:`np.ndarray`
+        The MSD of each individual particle with respect to lag-time.
+    n_frames : int
+        Number of frames in trajectory.
+    n_particles : int
+        Number of particles MSD was calculated over.
+
+    """
+
+    def __init__(self, u, select=None, msd_type='xyz', fft=True, **kwargs):
+        r"""
+        Parameters
+        ----------
+        u : Universe
+            An MDAnalysis :class:`Universe`.
+        selection : str
+            An MDAnalysis selection string. Defaults to `None`.
+        msd_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+            Desired dimensions to be included in the MSD. Defaults to 'xyz'.
+        fft : bool
+            Use a fast FFT based algorithm for computation of the MSD.
+            Otherwise, use the simple "windowed" algorithm. Defaults to ``True``.
+
+        """
+        self.u = u
+        
+        super(EinsteinMSD, self).__init__(self.u.trajectory, **kwargs)
+
+        #args
+        self.select = select
+        self.msd_type = msd_type
+        self.fft = fft
+        
+        #local
+        self._dim = None
+        self._position_array = None
+        self._atoms = None
+        
+        #indexing
+        self._n_frames = 0
+        self._n_particles = 0
+
+        #result
+        self.dim_fac = 0
+        self.timeseries = None
+        self.msds_by_particle = None
+
+    def _prepare(self):
+        self._parse_msd_type()
+        self._atoms = self.u.select_atoms(self.select)
+        self._n_frames = len(self.u.trajectory)
+        self._n_particles = len(self._atoms)
+        self._position_array = np.zeros((self._n_frames, self._n_particles, 3))
```

Comment:
For large trajectories, this can exhaust the memory. 

I think the docs should contain a note that this function might consume a large amount of memory.

Reply:
Being able to handle `step` (and `start` and `stop`) correctly gives a user a chance to control memory.

Reply:
As @orbeckst said, only make this as big as you need to. For example, you don't need all 3 dimensions if `msd_type='x'`. 

```suggestion
        self._position_array = np.zeros((self.n_frames, self._n_particles, self.dim_fac))
```